lib/segment/src/index/field_index/full_text_index/mmap_inverted_index/mmap_postings.rs

```diff
@@ -10,14 +12,98 @@ use memmap2::Mmap;
 use memory::madvise::{Advice, AdviceSetting, Madviseable};
 use memory::mmap_ops::open_read_mmap;
 use posting_list::{
-    IdsPostingList, IdsPostingListView, PostingChunk, PostingListComponents, RemainderPosting,
+    PostingChunk, PostingList, PostingListComponents, PostingListView, PostingValue,
+    RemainderPosting, ValueHandler,
 };
-use zerocopy::{FromBytes, Immutable, IntoBytes, KnownLayout};
+use zerocopy::{FromBytes, Immutable, IntoBytes, KnownLayout, Unaligned};
 
 use crate::index::field_index::full_text_index::inverted_index::TokenId;
+use crate::index::field_index::full_text_index::positions::Positions;
 
 const ALIGNMENT: usize = 4;
 
+pub type JustIds = ();
+
+/// Trait marker to enrich [`posting_list::PostingValue`] for handling mmap files with the posting list.
+pub(in crate::index::field_index::full_text_index) trait MmapPostingValue:
+    PostingValue<
+    Handler: ValueHandler<Sized: FromBytes + Immutable + IntoBytes + KnownLayout + Unaligned>
+                 + Clone
+                 + Debug,
+>
+{
+    /// Length in bytes of the variable size data section.
+    fn var_size_data_len(var_size_data: &<Self::Handler as ValueHandler>::VarSizeData) -> usize;
```

Comment:
As I understood from the call review, `VarSizeData` can either be `()` or `[u8]`, and if we always use `[u8]`, the price is that we spend extra 16bytes per posting list. But I think it is worth paying, cause otherwise complexity of the code is quite significant.

Reply:
simplified in https://github.com/qdrant/qdrant/pull/6619/commits/a06ab78081cc5a037af2528df5a98ba8bfd8d0a4